### PR TITLE
[Merged by Bors] - feat: golf using `module`/`match_scalars` throughout the library

### PR DESCRIPTION
**Pull request by hrmacbeth**
50 sample uses of the new (#16593) `module` and `match_scalars` tactics.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Analytic/Meromorphic.lean [hunk 1]
```diff
@@ -227,8 +227,8 @@ lemma iff_eventuallyEq_zpow_smul_analyticAt {f : 𝕜 → E} {x : 𝕜} : Meromo
     ∃ (n : ℤ) (g : 𝕜 → E), AnalyticAt 𝕜 g x ∧ ∀ᶠ z in 𝓝[≠] x, f z = (z - x) ^ n • g z := by
   refine ⟨fun ⟨n, hn⟩ ↦ ⟨-n, _, ⟨hn, eventually_nhdsWithin_iff.mpr ?_⟩⟩, ?_⟩
   · filter_upwards with z hz
-    rw [← mul_smul, ← zpow_natCast, ← zpow_add₀ (sub_ne_zero.mpr hz), neg_add_cancel,
-      zpow_zero, one_smul]
+    match_scalars
+    field_simp [sub_ne_zero.mpr hz]
```
Comment (grunweg) [inline Mathlib/Analysis/Analytic/Meromorphic.lean:231]: Note to self: benchmark this --- but this a clear improvement in terms of readability!

File: Mathlib/Analysis/Convex/Join.lean [hunk 2]
```diff
@@ -113,19 +113,16 @@ theorem convexJoin_assoc_aux (s t u : Set E) :
...
     ⟨x, hx, (a₂ * b₁ / (a₂ * b₁ + b₂)) • y + (b₂ / (a₂ * b₁ + b₂)) • z,
       ⟨y, hy, z, hz, _, _, ?_, ?_, ?_, rfl⟩,
-      a₂ * a₁, a₂ * b₁ + b₂, mul_nonneg ha₂ ha₁, hab.le, ?_, ?_⟩
-  · exact div_nonneg ha₂b₁ hab.le
-  · exact div_nonneg hb₂.le hab.le
-  · rw [← add_div, div_self hab.ne']
-  · rw [← add_assoc, ← mul_add, hab₁, mul_one, hab₂]
-  · simp_rw [smul_add, ← mul_smul, mul_div_cancel₀ _ hab.ne', add_assoc]
+      a₂ * a₁, a₂ * b₁ + b₂, by positivity, by positivity, ?_, ?_⟩
+  · positivity
```
Comment (grunweg) [inline Mathlib/Analysis/Convex/Join.lean:121]: Optional: you could also fill in (up to) three of the metavariables above already, as their proofs are merely `positivity` or `field_simp`.

File: Mathlib/Analysis/Convex/Strict.lean [hunk 3]
```diff
@@ -359,16 +358,14 @@ theorem strictConvex_iff_div :
     StrictConvex 𝕜 s ↔
       s.Pairwise fun x y =>
         ∀ ⦃a b : 𝕜⦄, 0 < a → 0 < b → (a / (a + b)) • x + (b / (a + b)) • y ∈ interior s :=
-  ⟨fun h x hx y hy hxy a b ha hb => by
-    apply h hx hy hxy (div_pos ha <| add_pos ha hb) (div_pos hb <| add_pos ha hb)
-    rw [← add_div]
-    exact div_self (add_pos ha hb).ne', fun h x hx y hy hxy a b ha hb hab => by
+  ⟨fun h x hx y hy hxy a b ha hb => h hx hy hxy (by positivity) (by positivity) (by field_simp),
```
Comment (grunweg) [inline Mathlib/Analysis/Convex/Strict.lean:361]: Small nit: replace by => by \mapsto, per the style guide. Not something I'd insist on.

**Review by grunweg — COMMENTED** [review]
This is very nice, thanks! I have three minor comments, which need not block this from landing.
maintainer merge

**Comment by grunweg** [comment]
!bench

**Comment by sgouezel** [comment]
bors d+

**Pull request closed without merging**